cli/info/info.go

```diff
@@ -24,29 +28,86 @@ Flags:
 
 var infoFlags = []string{"remote", "label", "profile", "config"}
 
-func infoMain(args []string, c cli.Config) (err error) {
-	if len(args) > 0 {
-		return goerr.New("argument is provided but not defined; please refer to the usage by flag -h.")
-	}
-
-	if c.Remote == "" {
-		return goerr.New("Remote is not given; please refer to the usage by flag -h.")
-	}
-
-	serv := client.NewServer(c.Remote)
+func getInfoFromRemote(c cli.Config) (resp *client.InfoResp, err error) {
 
 	req := new(client.InfoReq)
 	req.Label = c.Label
 	req.Profile = c.Profile
 
+	serv := client.NewServer(c.Remote)
+
 	reqJSON, _ := json.Marshal(req)
-	certPem, err := serv.Info(reqJSON)
+	resp, err = serv.Info(reqJSON)
+
+	_, err = helpers.ParseCertificatePEM([]byte(resp.Certificate))
 	if err != nil {
-		return err
+		return
 	}
-	resp := client.InfoResp{
-		Certificate: string(certPem),
+
+	return
+}
+
+func getInfoFromConfig(c cli.Config) (resp *client.InfoResp, err error) {
+	s, err := sign.SignerFromConfig(c)
+	if err != nil {
+		return
+	}
+
+	cert, err := s.Certificate(c.Label, c.Profile)
+	if err != nil {
+		return
 	}
+
+	blk := pem.Block{
+		Type:  "CERTIFICATE",
+		Bytes: cert.Raw,
+	}
+
+	certPem := pem.EncodeToMemory(&blk)
+
+	var profile *config.SigningProfile
+
+	policy := s.Policy()
+
+	if policy != nil && policy.Profiles != nil && c.Profile != "" {
+		profile = policy.Profiles[c.Profile]
+	}
+
+	if profile == nil && policy != nil {
+		profile = policy.Default
+	}
+
+	resp = &client.InfoResp{
+		Certificate:  string(certPem),
+		Usage:        profile.Usage,
+		ExpiryString: profile.ExpiryString,
+	}
+
+	return
+}
+
+func infoMain(args []string, c cli.Config) (err error) {
+	if len(args) > 0 {
+		return goerr.New("argument is provided but not defined; please refer to the usage by flag -h.")
+	}
+
+	var resp *client.InfoResp
+
+	if c.Remote != "" {
+		resp, err = getInfoFromRemote(c)
+		if err != nil {
+			return
+		}
+
+	} else if c.CFG != nil {
+		resp, err = getInfoFromConfig(c)
+		if err != nil {
+			return
+		}
+	} else {
+		return goerr.New("Either -remote or -config must be given. Do cfssl info -h for usage.")
```

Comment:
Instead of 'Do cfssl info -h', I would phrase it "Refer to cfssl info -h" or something along those lines.